platform/engineering/design-docs/2020-06-02-Sentry10.md

```diff
@@ -0,0 +1,142 @@
+sentry_onpremise_postgres_1                 docker-entrypoint.sh postgres    Up      5432/tcp                    
+sentry_onpremise_redis_1                    docker-entrypoint.sh redis ...   Up      6379/tcp                    
+sentry_onpremise_sentry-cleanup_1           /entrypoint.sh 0 0 * * * g ...   Up      9000/tcp                    
+sentry_onpremise_smtp_1                     docker-entrypoint.sh exim  ...   Up      25/tcp                      
+sentry_onpremise_snuba-api_1                ./docker_entrypoint.sh api       Up      1218/tcp                    
+sentry_onpremise_snuba-cleanup_1            /entrypoint.sh */5 * * * * ...   Up      1218/tcp                    
+sentry_onpremise_snuba-consumer_1           ./docker_entrypoint.sh con ...   Up      1218/tcp                    
+sentry_onpremise_snuba-replacer_1           ./docker_entrypoint.sh rep ...   Up      1218/tcp                    
+sentry_onpremise_symbolicator-cleanup_1     /entrypoint.sh 55 23 * * * ...   Up      3021/tcp                    
+sentry_onpremise_symbolicator_1             /bin/bash /docker-entrypoi ...   Up      3021/tcp                    
+sentry_onpremise_web_1                      /bin/sh -c exec /docker-en ...   Up      0.0.0.0:9000->9000/tcp      
+sentry_onpremise_worker_1                   /bin/sh -c exec /docker-en ...   Up      9000/tcp                    
+sentry_onpremise_zookeeper_1                /etc/confluent/docker/run        Up      2181/tcp, 2888/tcp, 3888/tcp
+```
+
+For comparison sake, our Sentry 9 uses 3 docker containers:
+
+- web
+- worker
+- cron
+
+
+### Code Location
+All code will live in the [`devops` repo](https://github.com/department-of-veterans-affairs/devops)
+
+### Testing Plan
+We will stand up Sentry 10 in parallel with our existing Sentry 9.1.2. We will then configure the `Settings.sentry.dsn` of `dev` and/or `staging` to use the Sentry 10 deployment.
+
+### Logging
+N/A
+
+### Debugging
+N/A - I think?
+
+### Caveats
+_To be determined._
+
+### Security Concerns
+No additional security concerns to the ones we have with our existing Sentry.
+
+### Privacy Concerns
+No additional privacy concerns to the ones we have with our existing Sentry.
+
+### Open Questions and Risks
+#### 1. Performance Scaling
+Since we intend on deploying to a single EC2 instance, there is not much room for scaling up in the event that Sentry 10 runs into CPU or memory bottlenecks.
+
+#### 2. Moving Away from `getsentry/onpremise` Happy Path
+The [`getsentry/onpremise` repo](https://github.com/getsentry/onpremise) is a great way to quickly run your own "on premise" Sentry 10 application. It was not designed with scale in mind. In this design, we will be heavily relying on this bootstrapping repo for building and deploying. It's likely that we will eventually have a need to do something custom that would require us to fork the `getsentry/onpremise` repo, making any future modifications and upgrades difficult.
+
```

Comment:
An issue with this method also is that a deployment of Sentry will result in downtime while the disk is unmounted and shuffled around between instances. I'm assuming that you know that already, but may be worth calling out here. We've done zero-downtime stuff for everything but those instances that need to shuffle disks around in the past and accepted it on those services, but that has not been the case for Sentry to date.

Reply:
I did not already realize that. That's a big problem. How is this handled for prometheus?